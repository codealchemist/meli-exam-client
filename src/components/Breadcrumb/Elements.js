import styled from 'styled-components'

export const Wrapper = styled.div`
  max-width: 1200px;
  width: 100%;
  margin: auto;
  padding: 20px 0;
  display: flex;
`

export const ContentWrapper = styled.div`
  display: flex;
  align-items: center;
  width: 100%;
  padding: 0 10px;
`

export const Content = styled.div`
  display: flex;
  align-items: center;

  i {
    font-style: normal;
    padding: 0 5px;
    display: flex;
    align-items: center;
  }

  div {
    cursor: pointer;
  }
`
