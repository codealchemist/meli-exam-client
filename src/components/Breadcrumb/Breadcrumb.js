import React from 'react'
import { Wrapper, ContentWrapper, Content } from './Elements'

const Breadcrumb = ({ categories }) => (
  <Wrapper>
    <ContentWrapper>
      {categories &&
        categories.map((category, index) => (
          <Content key={category.id}>
            {!!index && <i className="breadcrumb-chevron" />}
            <div>{category.name}</div>
          </Content>
        ))}
    </ContentWrapper>
  </Wrapper>
)

export default Breadcrumb
