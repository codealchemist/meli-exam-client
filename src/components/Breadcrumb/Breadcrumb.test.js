import React from 'react'
import { mount } from 'enzyme'
import Breadcrumb from './Breadcrumb'

describe('Breadcrumb', () => {
  it('should render empty component', () => {
    const categories = []
    const wrapper = mount(<Breadcrumb categories={categories} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render breadcrumbs', () => {
    const categories = [
      { id: 42, name: 'The answer to everything' },
      { id: 86, name: 'Maxwell Smart' }
    ]
    const wrapper = mount(<Breadcrumb categories={categories} />)
    expect(wrapper).toMatchSnapshot()
  })
})
