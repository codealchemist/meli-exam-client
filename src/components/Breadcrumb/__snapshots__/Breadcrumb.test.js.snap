// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb should render breadcrumbs 1`] = `
<Breadcrumb
  categories={
    Array [
      Object {
        "id": 42,
        "name": "The answer to everything",
      },
      Object {
        "id": 86,
        "name": "Maxwell Smart",
      },
    ]
  }
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "cxEVoG",
            "rules": Array [
              "
  max-width: 1200px;
  width: 100%;
  margin: auto;
  padding: 20px 0;
  display: flex;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bdVaJa cxEVoG"
      >
        <styled.div
          key="42"
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "lastClassName": "fuoGiD",
                  "rules": Array [
                    "
  display: flex;

  i {
    font-style: normal;
    padding: 0 5px;
    display: flex;
    align-items: center;
  }

  div {
    cursor: pointer;
  }
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-bwzfXH fuoGiD"
            >
              <div>
                The answer to everything
              </div>
            </div>
          </StyledComponent>
        </styled.div>
        <styled.div
          key="86"
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "lastClassName": "fuoGiD",
                  "rules": Array [
                    "
  display: flex;

  i {
    font-style: normal;
    padding: 0 5px;
    display: flex;
    align-items: center;
  }

  div {
    cursor: pointer;
  }
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-bwzfXH fuoGiD"
            >
              <i
                className="breadcrumb-chevron"
              />
              <div>
                Maxwell Smart
              </div>
            </div>
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</Breadcrumb>
`;

exports[`Breadcrumb should render empty component 1`] = `
<Breadcrumb
  categories={Array []}
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "cxEVoG",
            "rules": Array [
              "
  max-width: 1200px;
  width: 100%;
  margin: auto;
  padding: 20px 0;
  display: flex;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bdVaJa cxEVoG"
      />
    </StyledComponent>
  </styled.div>
</Breadcrumb>
`;
